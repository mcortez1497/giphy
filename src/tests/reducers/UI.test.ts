import {
  resetRegistration,
  setDrawerOpen,
  setDrawerClosed,
  setRegistrationComplete,
  uiReducer
} from "reducers";

const initialState = {
  didRegister: false,
  isDrawerOpen: false
};

describe("UI State", () => {
  describe("Actions", () => {
    it("returns resetRegistration action", () => {
      expect(resetRegistration()).toMatchSnapshot();
    });

    it("returns setDrawerOpen action", () => {
      expect(setDrawerOpen()).toMatchSnapshot();
    });

    it("returns setDrawerClosed action", () => {
      expect(setDrawerClosed()).toMatchSnapshot();
    });

    it("returns setRegistrationComplete action", () => {
      expect(setRegistrationComplete()).toMatchSnapshot();
    });
  });

  describe("Reducer", () => {
    it("resets registration", () => {
      const action = {
        type: "RESET_REGISTRATION"
      };
      expect(uiReducer(initialState, action)).toMatchSnapshot();
    });

    it("sets drawer open", () => {
      const action = {
        type: "SET_DRAWER_OPEN"
      };
      expect(uiReducer(initialState, action)).toMatchSnapshot();
    });

    it("sets drawer closed", () => {
      const action = {
        type: "SET_DRAWER_CLOSED"
      };
      expect(uiReducer(initialState, action)).toMatchSnapshot();
    });

    it("sets registration complete", () => {
      const action = {
        type: "SET_REGISTRATION_COMPLETE"
      };
      expect(uiReducer(initialState, action)).toMatchSnapshot();
    });
  });
});
