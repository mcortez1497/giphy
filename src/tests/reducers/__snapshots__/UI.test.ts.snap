// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI State Actions returns resetRegistration action 1`] = `
Object {
  "type": "RESET_REGISTRATION",
}
`;

exports[`UI State Actions returns setDrawerClosed action 1`] = `
Object {
  "type": "SET_DRAWER_CLOSED",
}
`;

exports[`UI State Actions returns setDrawerOpen action 1`] = `
Object {
  "type": "SET_DRAWER_OPEN",
}
`;

exports[`UI State Actions returns setRegistrationComplete action 1`] = `
Object {
  "type": "SET_REGISTRATION_COMPLETE",
}
`;

exports[`UI State Reducer resets registration 1`] = `
Object {
  "didRegister": false,
  "isDrawerOpen": false,
}
`;

exports[`UI State Reducer sets drawer closed 1`] = `
Object {
  "didRegister": false,
  "isDrawerOpen": false,
}
`;

exports[`UI State Reducer sets drawer open 1`] = `
Object {
  "didRegister": false,
  "isDrawerOpen": true,
}
`;

exports[`UI State Reducer sets registration complete 1`] = `
Object {
  "didRegister": true,
  "isDrawerOpen": false,
}
`;
